src/legacy/core_plugins/kibana/ui_setting_defaults.js

```diff
@@ -300,19 +320,21 @@ export function getUiSettingDefaults() {
       }),
       value: 'sessionId',
       options: ['sessionId', 'custom', 'none'],
+      optionLabels: requestPreferenceOptionLabels,
       type: 'select',
       description: i18n.translate('kbn.advancedSettings.courier.requestPreferenceText', {
         defaultMessage:
           `Allows you to set which shards handle your search requests.
           <ul>
-            <li><strong>sessionId:</strong> restricts operations to execute all search requests on the same shards.
+            <li><strong>{sessionId}:</strong> restricts operations to execute all search requests on the same shards.
               This has the benefit of reusing shard caches across requests.</li>
-            <li><strong>custom:</strong> allows you to define a your own preference.
+            <li><strong>{custom}:</strong> allows you to define a your own preference.
               Use <strong>courier:customRequestPreference</strong> to customize your preference value.</li>
-            <li><strong>none:</strong> means do not set a preference.
+            <li><strong>{none}:</strong> means do not set a preference.
               This might provide better performance because requests can be spread across all shard copies.
               However, results might be inconsistent because different shards might be in different refresh states.</li>
           </ul>`,
+        values: requestPreferenceOptionLabels,
```

Comment:
```suggestion
        values: {
            sessionId: i18n.translate('kbn.advancedSettings.courier.requestPreferenceSessionId', {
              defaultMessage: 'Session ID',
            }),
            custom: i18n.translate('kbn.advancedSettings.courier.requestPreferenceCustom', {
              defaultMessage: 'Custom',
            }),
            none: i18n.translate('kbn.advancedSettings.courier.requestPreferenceNone', {
              defaultMessage: 'None',
            }),
          },
```

This will solve the CI. You cannot use variables inside the i18n translations strings